src/js/actions/mask.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2015 Adobe Systems Incorporated. All rights reserved.
+ *  
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var Promise = require("bluebird");
+
+    var layerActions = require("./layers"),
+        menuActions = require("./menu"),
+        toolActions = require("./tools"),
+        locks = require("../locks");
+
+    var _CLEAR_PATH = 106;
+
+    /**
+     * Handle the deleteText for vector mask mode tools
+     *
+     * @return {Promise}
+     */
+    var handleDelete = function () {
+        var appStore = this.flux.store("application"),
+            currentDocument = appStore.getCurrentDocument();
+
+        if (!currentDocument) {
+            return Promise.resolve();
+        }
+
+        var currentLayer = currentDocument.layers.selected.first();
+
+        return this.transfer(layerActions.resetLayers, currentDocument, currentLayer)
```

Comment:
Sure, 
We dont have a model for the Paths that photoshop is currently interacting with. If we had a notificaiton system in place that could tell us if the current mask was empty etc we wouldn't need to do this ( and should probably be logged on a backlog the implement), but since we don't have this we need to differentiate between the various cases of delete. namely selected component, no selected component, and empty mask. 

the first reset differentiates between the empty mask and non empty mask. 
the second reset differentiates between the non empty mask with and without selected components.